pkg/network/tracer/tracer_linux_test.go

```diff
@@ -2068,3 +2069,67 @@ func TestBlockingReadCounts(t *testing.T) {
 
 	assert.Equal(t, uint64(n), conn.MonotonicSum().RecvBytes)
 }
+
+// GoTLS test
+
+func TestHTTPGoTLSCaptureNewProcess(t *testing.T) {
+	const ClientBin string = "./testutil/cmd/gotls_client/gotls_client"
+	const ServerAddr string = "localhost:8081"
+	const ExpectedOccurrences int = 10
+
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	if !httpsSupported(t) {
+		t.Skip("HTTPS feature not available supported for this setup")
+	}
+
+	// Given
+	var closeServer func() = httptest.HTTPServer(t, ServerAddr, httptest.Options{
+		EnableTLS: true,
+	})
+
+	cfg := config.New()
+	cfg.EnableHTTPMonitoring = true
+	cfg.EnableHTTPSMonitoring = true
+	cfg.BPFDebug = true
+
+	tr, err := NewTracer(cfg)
+	require.NoError(t, err)
+	defer tr.Stop()
+	require.NoError(t, tr.RegisterClient("1"))
+
+	// When
+	req, err := nethttp.NewRequest(nethttp.MethodGet, fmt.Sprintf("https://%s/%d/request", ServerAddr, nethttp.StatusOK), nil)
+	require.NoError(t, err)
+
+	clientBuildCmd := fmt.Sprintf("go build -o %s ./testutil/cmd/gotls_client", ClientBin)
+	clientCmd := fmt.Sprintf("%s %s %d", ClientBin, ServerAddr, ExpectedOccurrences)
+	t.Cleanup(func() { os.RemoveAll(ClientBin) })
+	_ = testutil.RunCommands(t, []string{clientBuildCmd, clientCmd}, false)
+
+	closeServer()
+
+	// Then
+	occurrences := 0
+	require.Eventually(t, func() bool {
+		stats, err := tr.GetActiveConnections("1")
+		require.NoError(t, err)
+		occurrences += countRequestOccurrences(t, stats, req)
+		return occurrences == ExpectedOccurrences
+	}, 3*time.Second, 100*time.Millisecond, "Expected to find the request %v times, got %v captured", ExpectedOccurrences, occurrences)
+}
+
+func countRequestOccurrences(t *testing.T, conns *network.Connections, req *nethttp.Request) (occurrences int) {
+	t.Helper()
+
+	expectedStatus := httptest.StatusFromPath(req.URL.Path)
+	for key, stats := range conns.HTTP {
+		if key.Path.Content == req.URL.Path && stats.HasStats(expectedStatus) {
+			occurrences++
+		}
+	}
+
+	return
+}
```

Comment:
there is a great deal to understand which request we didn't capture
for example, never capturing the last request can indicate a problem with "tls shutdown" mechanism
I do recommend to mimic the behavior from `TestOpenSSLVersions` and help us understand which requests we have missed